src/Devito.jl

```diff
@@ -799,14 +806,14 @@ function subdomain(name::String, instructions...)
         function define(self, dimensions)
             dims = Dict()
             for idim = 1:length(dimensions)
-                dims[dimensions[idim]] = self.instructions[idim]
+                dims[dimensions[idim]] = self.instructions[idim] == (nothing,) ? dimensions[idim] : self.instructions[idim]
```

Comment:
```suggestion
                dims[dimensions[idim]] = self.instructions[idim] == ("middle",0,0) ? dimensions[idim] : self.instructions[idim]
```
This seems simple to understand, and then you can remove the `instructiontest` function :)